tests/cluster/etl/test_loader.py

```diff
@@ -9,41 +9,100 @@
 
 class LoaderTests(unittest.TestCase):
     @classmethod
-    def setUpClass(cls):
+    def setUp(self):
+
+        self.loader = Loader()
+        self.loader.save = MagicMock()
+        self.loader.save_many = MagicMock()
+
+        self.loader_input_key = Loader(dataset_input_key="my_key_1")
+        self.loader_input_key.save = MagicMock()
+        self.loader_input_key.save_many = MagicMock()
+
+        self.dataset_input_key_list = ["my_key_1", "my_key_2"]
+        self.loader_input_key_list = Loader(
+            dataset_input_key_list=self.dataset_input_key_list
+        )
+        self.loader_input_key_list.save = MagicMock()
+        self.loader_input_key_list.save_many = MagicMock()
+
+        self.df_1 = create_dataframe()
+        self.df_2 = create_dataframe()
+        self.df_3 = create_dataframe()
+
+    def test_return_inputs(self):
+        # Assert Loader returns ouput
+        input = {"my_key_1": self.df_1}
+        result = self.loader.etl(input)
+        self.assertIs(input, result)
 
-        cls.loader = Loader()
-        cls.loader.save = MagicMock()
-        cls.loader.save_many = MagicMock()
+    def test_save(self):
+        # save is called when:
+        # - inputs has one df
+        # - dataset_input_key is not given
+        self.loader.etl({"my_key_1": self.df_1})
 
-        cls.df = create_dataframe()
+        self.assertTrue(self.loader.save.called)
+        self.assertFalse(self.loader.save_many.called)
 
-    def test_save_single(self):
+        args = self.loader.save.call_args[
+            0
+        ]  # the .args part became available in python 3.8
+        self.assertEqual(len(args), 1)
+        self.assertIs(args[0], self.df_1)
 
-        input = {"myset": self.df}
-        result = self.loader.etl(input)
+    def test_save_many(self):
+        # save is called when:
+        # - inputs has more than one df
+        # - dataset_input_key is given
+        self.loader.etl({"my_key_1": self.df_1, "my_key_2": self.df_2})
 
-        # assert that loader returns something:
-        self.assertIs(input, result)
+        self.assertFalse(self.loader.save.called)
+        self.assertTrue(self.loader.save_many.called)
 
-        args = self.loader.save.call_args[
+        args = self.loader.save_many.call_args[
             0
         ]  # the .args part became available in python 3.8
         print(args)
```

Comment:
is it neccesary to print the args?

Reply:
No, it was just already there, i will remove.